### 1. Data loading

In [24]:
import pandas as pd
import numpy as np
from math import sqrt
from scipy import stats

control_df = pd.read_csv("./control_group.csv", sep=";")
test_df = pd.read_csv("./test_group.csv", sep=";")


control_df.head()

,Campaign Name,Date,Spend [USD],# of Impressions,Reach,# of Website Clicks,# of Searches,# of View Content,# of Add to Cart,# of Purchase
0,Control Campaign,1.08.2019,2280,82702.0,56930.0,7016.0,2290.0,2159.0,1819.0,618.0
1,Control Campaign,2.08.2019,1757,121040.0,102513.0,8110.0,2033.0,1841.0,1219.0,511.0
2,Control Campaign,3.08.2019,2343,131711.0,110862.0,6508.0,1737.0,1549.0,1134.0,372.0
3,Control Campaign,4.08.2019,1940,72878.0,61235.0,3065.0,1042.0,982.0,1183.0,340.0
4,Control Campaign,5.08.2019,1835,NaN,NaN,NaN,NaN,NaN,NaN,NaN


In [25]:
test_df.head()

,Campaign Name,Date,Spend [USD],# of Impressions,Reach,# of Website Clicks,# of Searches,# of View Content,# of Add to Cart,# of Purchase
0,Test Campaign,1.08.2019,3008,39550,35820,3038,1946,1069,894,255
1,Test Campaign,2.08.2019,2542,100719,91236,4657,2359,1548,879,677
2,Test Campaign,3.08.2019,2365,70263,45198,7885,2572,2367,1268,578
3,Test Campaign,4.08.2019,2710,78451,25937,4216,2216,1437,566,340
4,Test Campaign,5.08.2019,2297,114295,95138,5863,2106,858,956,768


### 2. Data preparation

The control group dataset contains a day when advertising costs were recorded, but there are no statistics for this day at all, indicating a technical failure. Decision is to remove this day from the control and test groups.

In [26]:
control_clean = control_df.dropna(subset=["# of Purchase"]).copy()
test_clean = test_df.drop(4).copy()

### 3. Data aggregation

In [27]:
control_atc = int(control_df["# of Add to Cart"].sum())
control_buy = int(control_df["# of Purchase"].sum())
test_atc = int(test_df["# of Add to Cart"].sum())
test_buy = int(test_df["# of Purchase"].sum())

### 4. Metrics calculation

Delta conversion

In [32]:
conversion_test = test_buy / test_atc
conversion_control = control_buy / control_atc

conversion_test, conversion_control

(0.5912803448536641, 0.40214854111405834)

In [29]:
delta = conversion_test - conversion_control
delta

0.18913180373960575

Standard error for difference in proportions

In [30]:
se = sqrt(conversion_test * (1 - conversion_test) / test_atc + conversion_control * (1 - conversion_control) / control_atc)
se

0.003938970411860796

95% confidence interval

In [31]:
z = stats.norm.ppf(0.975)
ci_low = delta - z * se
ci_high = delta + z * se

ci_low, ci_high

(np.float64(0.18141156359618968), np.float64(0.19685204388302183))

### 5. Summary

From the previous task, it is clear that the “Add-to-Cart → Purchase” conversion for the test group is 59.13%, and for the control group it is 40.21%. The difference is 18.91%. 95% CI is from 18.14% to 19.69%. Std error = 0.0039.